.github/workflows/deploy.yml

```diff
@@ -0,0 +1,55 @@
+name: Deploy
+
+# We deploy the site *only* on
+#
+# 1. A push to master, or
+# 2. A manually triggered Deploy build
+#
+# We don't want to deploy for every branch and PR of course!
+on:
+  push:
+    branches:
+      - master
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      # Should probably do this with Nix
+      - name: Install lftp
+        run: sudo apt-get install lftp
```

Comment:
Yep, we can do this with Nix. To use the same patter as we did for `linkchecker`—seems reasonable—we would need to add `lftp` as an attribute to the output of `default.nix` then use the following command:

```
nix run -f . lftp -c lftp <lftp options>
```

We could also pass the `--quiet` flag to `nix run` if we want to suppress Nix's own output:

```
nix run --quiet -f . lftp -c lftp <...>
```

The `.` refers to the `default.nix` file at the top level of the repo; to run from a different directory, we would need to change that path.

I haven't tested this on GitHub Actions, but it worked locally:

```
>nix run -f . lftp -c lftp --help
Usage: lftp [OPTS] <site>
`lftp' is the first command executed by lftp after rc files
 -f <file>           execute commands from the file and exit
...
```

I'll push a commit with these changes; if it doesn't work, we can always revert it :)

Reply:
Actually, not sure if I can just push a commit to ` tomjaguarpaw:master`. If that doesn't work, I might just leave the changes as GitHub suggestions since they're pretty small.

Reply:
Down the line, we'll probably want to organize this a bit better—something like a `ci.nix` file that has all the tools and dependencies we need for CI, so that we can manage everything from one place and don't have to use `nix run` all the time. For now I'll go with the quicker solution, but this is something we should think of refactoring, maybe the next time we want to add a new tool or dependency to CI.

Reply:
Okay, see [this PR](https://github.com/tomjaguarpaw/www.haskell.org/pull/1) for the changes :)

Reply:
I incorporated your change to get `lftp` via nix, thanks!